tests/foreman/cli/test_repository.py

```diff
@@ -602,14 +602,16 @@ def test_positive_create_yum_repo_with_checksum_type(self):
         :CaseImportance: Critical
         """
         for checksum_type in u'sha1', u'sha256':
-            with self.subTest(checksum_type):
-                content_type = u'yum'
-                repository = self._make_repository({
-                    u'checksum-type': checksum_type,
-                    u'content-type': content_type,
-                })
-                self.assertEqual(repository['content-type'], content_type)
-                self.assertEqual(repository['checksum-type'], checksum_type)
+            for downloadpolicy in u'background', u'immediate':
```

Comment:
that test case is talking specifically for checksum functionality, you can separate your idea into two test cases

Reply:
@oshtaier earlier the test was creating the repo with default download_policy (on_demand) but not covering the other type of download_policies i.e. background/immediate hence added the same to cover more on it.
and we added the nagative test for on_demand separately to cover on that type of download policy .

Do you think here should be two test one per download policy background/immediate?

Reply:
okay.